azurerm/internal/services/containers/container_group_resource.go

```diff
@@ -597,6 +598,39 @@ func resourceArmContainerGroupCreate(d *schema.ResourceData, meta interface{}) e
 	return resourceArmContainerGroupRead(d, meta)
 }
 
+func resourceArmContainerGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Containers.GroupsClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := containerinstance.Resource{
+		Tags: tags.Expand(t),
+	}
+
+	_, err := client.Update(ctx, resGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error updating container group %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read container group %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
```

Comment:
since we're performing an update here, we can remove this ID update (and Get retrieval) here, since the ID won't change during an update